backend/src/backend.cpp

```diff
@@ -27,53 +15,35 @@ class DroneCoreBackend::Impl
     Impl() {}
     ~Impl() {}
 
-    bool run(const int mavlink_listen_port)
+    void connect(const int mavlink_listen_port)
     {
         _connection_initiator.start(_dc, 14540);
         _connection_initiator.wait();
+    }
 
-        grpc::ServerBuilder builder;
-        setup_port(builder);
-
-        CoreServiceImpl<> core(_dc);
-        builder.RegisterService(&core);
-
-        Action action(_dc.system());
-        ActionServiceImpl<> action_service(action);
-        builder.RegisterService(&action_service);
-
-        Mission mission(_dc.system());
-        MissionServiceImpl<> mission_service(mission);
-        builder.RegisterService(&mission_service);
-
-        Telemetry telemetry(_dc.system());
-        TelemetryServiceImpl<> telemetry_service(telemetry);
-        builder.RegisterService(&telemetry_service);
-
-        _server = builder.BuildAndStart();
-        LogInfo() << "Server started";
-        _server->Wait();
-
-        return true;
+    void startGRPCServer()
+    {
+        _server = std::unique_ptr<GRPCServer>(new GRPCServer(_dc));
+        _server->run();
     }
 
-private:
-    void setup_port(grpc::ServerBuilder &builder)
+    void wait()
     {
-        std::string server_address("0.0.0.0:50051");
-        builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
-        LogInfo() << "Server set to listen on " << server_address;
+        _server->wait();
     }
 
-    dronecore::DroneCore _dc;
-    dronecore::backend::ConnectionInitiator<dronecore::DroneCore> _connection_initiator;
-    std::unique_ptr<grpc::Server> _server;
+private:
+    DroneCore _dc;
```

Comment:
Not reference ? I don't know, I'm just asking.

Reply:
No, because `backend.cpp` is actually the one owning the object. It has to be created somewhere, right? :-)

Reply:
Yes :)